FLUID-5249, FLUID-4884, FLUID-5506, FLUID-5587, FLUID-5264, FLUID-5616, FLUID-5495

amb26 opened the pull request:
Implementation of "global instantiator model" for framework, new "contextAwareness" implementation replacing final use case for demands blocks, and abolition of numerous obsolete framework features and their uses, including - 
 - demands blocks
 - the "old ChangeApplier",
 - "init functions" such as preInit, postInit and finalInit,
 - the "static environment"
 - "grade linkage system"
 - manual subcomponent instantiation, including initSubcomponent, etc.
 - "returnedPath" system
 - obsolete options syntax including @1, @2, COMPONENT_OPTIONS, etc.
 - others
Initial implementation of "IoC debugging system" capable of inspecting all view-bound grades in an Infusion context (browser pane). Preparing to abolish "eventedComponent" as a separate grade by moving all its materials to "littleComponent". Implementation of "priority constraint system" completing FLUID-5506, allowing "relative priority constraints" of the form "before:namespace", "after:namespace", etc.

colinbdclark commented on src/framework/core/js/DataBinding.js:
```diff
@@ -417,8 +439,8 @@ var fluid_2_0 = fluid_2_0 || {};
-        var transRec = instantiator.modelTransactions[transId];
-        if (!transRec && !instantiator.free) {
+        var transRec = instantiator && instantiator.modelTransactions[transId];
+        if (!transRec && instantiator) {
```
We check <code>instantiator</code> twice here. It looks like this could be rewritten as:

    if (instantiator && !instantiator.modelTransactions[transId]) { ... }

amb26 replied:
If the statement were written in that form, we'd then have to duplicate the reference expression to assign a value to transRec which is used in the if statement body and return value

colinbdclark replied:
Can you bail early in the case of <code>!instantiator</code> instead?

colinbdclark commented on demos/renderer/js/rendererDemo.js:
```diff
@@ -204,7 +203,7 @@ var demo = demo || {};
      */
     var setupDataModel = function () {
-        var applier = fluid.makeChangeApplier(demo.data);
+        var applier = fluid.makeHolderChangeApplier({model: demo.data});
```
Arguably orthogonal to this pull request, but this Renderer demo is immensely complex and full of code. Presumably a new demo is order some day.

amb26 replied:
Along with a new renderer :)

colinbdclark commented on demos/reorderer/gridReorderer/js/gridReordererDemo.js:
```diff
@@ -10,23 +10,20 @@ You may obtain a copy of the ECL 2.0 License and BSD License at
 (function (){
     "use strict";
+    
+    fluid.defaults("demo.initGridReorderer", {
```
Is <code>initGridReorderer</code> still the appropriate name for this now that it has been defined as a component grade?

colinbdclark commented on src/components/pager/js/Pager.js:
```diff
@@ -514,6 +513,8 @@ var fluid_2_0 = fluid_2_0 || {};
         },
         invokers: {
             acquireDefaultRange: {
+                // TODO: problem here - these are dynamic components and so cannot be constructed gingerly
```
Which "these" is this comment referring to? What does the expression "current (old) framework" refer to specifically?

colinbdclark commented on src/components/reorderer/js/ModuleLayout.js:
```diff
@@ -165,6 +165,13 @@ var fluid_2_0 = fluid_2_0 || {};
                     func: "{that}.computeLayout"
                 }
+            },
+            getRelativePosition: { // an old-fashioned function member
```
It's worth noting here why <code>getRelativePosition</code> is stuck as an "old-fashioned function member" instead of an invoker, and, if possible, what will need to be done in the future to fix this.

colinbdclark commented on src/components/undo/js/Undo.js:
```diff
@@ -169,10 +175,10 @@ var fluid_2_0 = fluid_2_0 || {};
         }
     });
-
+    
```
Stray whitespace.

colinbdclark commented on src/components/uploader/js/DemoUploadManager.js:
```diff
@@ -16,9 +16,17 @@ var fluid_2_0 = fluid_2_0 || {};
 (function ($, fluid) {
     "use strict";
-
+    
```
Stray whitespace.

colinbdclark commented on src/components/uploader/js/FileQueueView.js:
```diff
@@ -308,9 +309,13 @@ var fluid_2_0 = fluid_2_0 || {};
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewComponent"],
+        mergePolicy: {
+            //"members.queueFiles": "nomerge"
```
Is this here because of the commented-out <code>queueFiles</code> member below and the attached TODO? Is there another way to reflect the needed modelification of this component, rather than leaving commented-out code in?

colinbdclark commented on src/components/uploader/js/HTML5UploaderSupport.js:
```diff
@@ -14,22 +14,24 @@ var fluid_2_0 = fluid_2_0 || {};
+        }
     });
-
+    
```
Stray whitespace.

colinbdclark commented on src/framework/core/js/DataBinding.js:
```diff
@@ -1362,471 +1351,4 @@ var fluid_2_0 = fluid_2_0 || {};
     };
 
-
-    /** OLD CHANGEAPPLIER IMPLEMENTATION (Infusion 1.5 and before - this will be removed on Fluid 2.0) **/
```
Hurray!

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -685,6 +684,41 @@ var fluid = fluid || fluid_2_0;
+     * on small array sizes.
+     */
+    
+    fluid.stableSort = function (array, func) {
```
Is this a public-facing API? If so, can you document it?

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -818,6 +850,24 @@ var fluid = fluid || fluid_2_0;
         return fluid.makeArray(arguments).join(".");
     };
+    
+    function lastDotIndex(path) {
```
Is this not in a public scope for performance reasons?

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1015,11 +1074,138 @@ var fluid = fluid || fluid_2_0;
     var fluid_guid = 1;
 
-    /** Allocate an string value that will be very likely unique within this Fluid scope (frame or process) **/
+    /** Allocate an string value that will be very likely unique within this Infusion instance (frame or process) **/
```
Typo: "an string."

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1015,11 +1074,138 @@ var fluid = fluid || fluid_2_0;
+    
+    // Fluid priority system for encoding relative positions of, e.g. listeners, transforms, options, in lists
+    
+    fluid.extremePriority = 4e9; // around 2^32 - allows headroom of 21 fractional bits for sub-priorities
```
![Extreme surfing](http://www.bhmpics.com/walls/extreme_surfing_2-wide.jpg)
So extreme!

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1015,11 +1074,138 @@ var fluid = fluid || fluid_2_0;
+        if (recA.priority.fixed !== null && recB.priority.fixed !== null) {
+            return recA.priority.fixed - recB.priority.fixed;
+        } else { // sort constraint records to the end
+            return (recA.priority.fixed === null) - (recB.priority.fixed === null);
```
Some seriously tricky boolean coercion here. A comment will help.

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1545,13 +1721,13 @@ var fluid = fluid || fluid_2_0;
         return index;
     };
-
+    
```
Whitespace, yo.

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1607,7 +1786,7 @@ var fluid = fluid || fluid_2_0;
     fluid.makeComponents = function (components) {
```
Can we either mark this as unsupported, document it, or move it to a testing utility?

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1752,8 +1930,7 @@ var fluid = fluid || fluid_2_0;
     // to protect against circular evaluation
-    fluid.inEvaluationMarker = {"__CURRENTLY_IN_EVALUATION__": true};
-    fluid.destroyedMarker = {"__COMPONENT_DESTROYED__": true};
+    fluid.inEvaluationMarker = Object.freeze({"__CURRENTLY_IN_EVALUATION__": true});
```
:snowflake: :snowboarder: :icecream: :snowman:

colinbdclark commented on src/framework/core/js/Fluid.js:
```diff
@@ -1775,14 +1952,18 @@ var fluid = fluid || fluid_2_0;
-            else { // This is hardwired here for performance reasons - no need to protect deeper strategies
-                target[name] = fluid.inEvaluationMarker;
+            else {
+                if (target !== fluid.inEvaluationMarker) { // blatant "coding to the test" - this enables the simplest "re-trunking" in
```
Minor detail: this should be a TODO comment.

colinbdclark commented on src/framework/core/js/FluidView.js:
```diff
@@ -18,9 +18,9 @@ var fluid_2_0 = fluid_2_0 || {};
-    // The base grade for fluid.viewComponent and fluid.viewRelayComponent - will be removed again once the old ChangeApplier is eliminated
-    fluid.defaults("fluid.commonViewComponent", {
-        gradeNames: ["fluid.littleComponent", "autoInit"],
+    // The base grade for fluid.viewComponent and fluid.viewComponent - will be removed again once the old ChangeApplier is eliminated
```
This comment is out of date.

colinbdclark commented on src/framework/core/js/FluidView.js:
```diff
@@ -51,9 +41,9 @@ var fluid_2_0 = fluid_2_0 || {};
     // It preserves the current framework behaviour for the 1.4 release, but provides a more informative
     // diagnostic - in fact, it is perfectly acceptable for a component's creator to return no value and
-    // the failure is really in assumptions in fluid.initComponent. Revisit this issue for 1.5
+    // the failure is really in assumptions in fluid.initLittleComponent. Revisit this issue for 1.5
```
This comment is also out of date.

amb26 commented:
Thanks @colinbdclark - all review comments so far believed addressed

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+    fluid.defaults("fluid.debug.highlighter", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            highlightRoot: "#fluid-debug-highlightRoot"
```
Is there a reason why this selector is an ID instead of the standard Fluidic convention of nsamespaced classess?

amb26 replied:
This container must be unique within the entire document - since it is used to house all highlight nodes even which are destined to hover over elements within nested iframes.

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+        return "highlight-for:" + domId;
+    };
+    
+    fluid.debug.highlighter.highlight = function (that, highlightRoot, dispositions) {
```
"One function, one responsibility."

This could benefit from being split up into separate functions for creating, styling, and scrolling the highlight element.

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+        return output;
+    };
+    
+    fluid.debug.renderDefaults = function (typeName, options) {
```
Any reason not enable greater configurability by using stringTemplate() here instead of hardcoding the rendering of default blocks?

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+        if (fluid.isPrimitive(source)) {
+            if (typeof(source) === "string" && source.indexOf(options.findString) !== -1) {
+                var path = segs.slice(0, 2);
+                fluid.set(options.target, path, fluid.copy(fluid.get(options.fullSource, path)));
```
Let's assign the value of fluid.get() to a variable before copying and setting it.

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+            target: target,
+            fullSource: options
+        });
+        var markup = "<pre>" + JSON.stringify(target, null, 4) + "</pre>";
```
Again, hardcoded markup.

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+        rowIdToDomId[row.rowId] = row.domId;
+    };
+    
+    fluid.debug.renderOneDisposition = function (disp, indexElTemplate, defaultsIdToContent, rowIdToDomId) {
```
This function is quite complex and will also benefit from being split up.

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+        var allRows = [], rowIdToDomId = {};
+        fluid.each(dispositions, function (disp) {
+            var rows = fluid.debug.renderOneDisposition(disp, indexElTemplate, defaultsIdToContent, rowIdToDomId);
+            allRows = allRows.concat(rows); // yeah yeah we dun't have no mapcat
```
yeah we dun do!

<code>fluid.flatten</code>

colinbdclark commented on src/framework/core/js/FluidViewDebugging.js:
```diff
@@ -0,0 +1,649 @@
+                    styles: {
+                        tooltip: "fl-debug-tooltip"
+                    },
+                    position: {
```
Are these ultimately human-readable strings?

amb26 replied:
Typo - should read "center" but in fact was being ignored due to this being the default in any case - http://api.jqueryui.com/position/

colinbdclark commented on src/framework/enhancement/js/ContextAwareness.js:
```diff
@@ -0,0 +1,253 @@
+    };
+
+    /**
+     * Takes an hash of check context names to check records, designating a collection of context markers which might be registered at a location
```
I'd prefer that we avoid the term "hash" here (and else where in this file) in favour of "object" or "structure" or "specification"--perhaps even consistently naming the this kind of specification explicitly--"contextCheckSpec" or something like that. As a new reader (especially native JavaScripters), the "hash" terminology is confusing.

I can imagine that our documentation, in the future, may be quite careful about defining these types of declarative specifications up front (e.g. [we do a nice job for invoker specs](http://docs.fluidproject.org/infusion/development/Invokers.html#standard-invoker-binding-to-a-function-using-funcname-func) or whatever they should be called).

amb26 replied:
I think that the term "hash" is fairly well attested in the JavaScript community - e.g. http://www.quirksmode.org/js/associative.html , http://api.prototypejs.org/language/Hash/ , http://javascript.crockford.com/survey.html - and I don't think that the alternatives that you suggest (with the exception of "object") will be any more recognisable to the reader - and "object" carries the standard risk of confusion with "Objects derived from a prototype". 

The documentation for these features should be complete at https://github.com/fluid-project/infusion-docs/pull/63/files#diff-ea4f94ea49b20f80618f120626c0733fR19

colinbdclark replied:
The point is that you don't use the term consistently; have a look at other API documentation comments throughout the framework, and you'll notice the term "hash" is not widely used (and, with the exception of the <code>fluid.arrayToHash</code> function, has been largely introduced through this particular pull request).

Your point about the ambiguity of the term "object" actually supports my argument that we need to distinguish these clearly as a type of data structure used throughout the framework.

Regardless, you need to pick a term and apply it consistently throughout framework. Keep in mind the prevalence of the <code>{Object}</code> notation throughout your JSDoc-style comments. :wink:

colinbdclark commented on src/framework/enhancement/js/ProgressiveEnhancement.js:
```diff
@@ -29,116 +26,15 @@ var fluid_2_0 = fluid_2_0 || {};
-                funcName: "fluid.progressiveChecker.check",
-                args: ["{that}.options.checks", "{that}.options.defaultContextName"]
-            }
+    fluid.contextAware.makeChecks({
```
Given that we've moved checks like <code>isBrowser</code> up to ContextAwareness.js, and these checks are pretty uncommon now (all modern browsers implement XHR 2.0), should they be moved into the Uploader instead?

colinbdclark commented on src/framework/renderer/js/RendererUtilities.js:
```diff
@@ -99,12 +104,12 @@ fluid_2_0 = fluid_2_0 || {};
         mergePolicy: {
             "rendererOptions.idMap": "nomerge",
-            "rendererOptions.model": "preserve",
+            //"rendererOptions.model": "preserve",
```
Stray commented out code.

amb26 replied:
This reminded me that there is still a perplexing piece of rubbish to deal with in FluidIoC.js expandCompactRec line 1831, which is the sole remaining reference to the now abolished mergePolicy of "preserve". Dead and unlamented.

colinbdclark commented on src/lib/json/js/json2.js:
```diff
@@ -1,264 +0,0 @@
-/*
-    json2.js
```
:tada: :+1:

colinbdclark commented on tests/framework-tests/core/js/DataBindingTests.js:
```diff
@@ -2014,10 +1639,52 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        }
+    });
+    
+    fluid.tests.isNaN = function (value) { // replicate for Number.isNaN which is not supported on IE
```
Minor detail: Number.isNaN is also [not supported on Safari](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#Browser_compatibility).

colinbdclark commented on tests/framework-tests/core/js/DataBindingTests.js:
```diff
@@ -2014,10 +1639,52 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     
     fluid.defaults("fluid.tests.fluid5659relay", {
-        gradeNames: ["fluid.modelRelayComponent", "autoInit"],
+        gradeNames: ["fluid.modelComponent", "autoInit"],
```
Stray <code>autoInit</code>

colinbdclark commented on tests/framework-tests/core/js/DataBindingTests.js:
```diff
@@ -2112,6 +1779,6 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     };
     
-    fluid.test.runTests(["fluid.tests.fluid5659root"]);
+    // fluid.test.runTests(["fluid.tests.fluid5659root"]);
```
Why is this commented out?

amb26 replied:
Ignorance, Madam, pure ignorance!
http://www.todayinliterature.com/print-today.asp?Event_Date=8/16/1762

colinbdclark commented on tests/framework-tests/core/js/ModelTransformationTests.js:
```diff
@@ -2085,13 +2084,13 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     fluid.makeComponents({
-        "farm.cat": "fluid.littleComponent",
-        "bowl.fish": "fluid.littleComponent"
+        "farm.cat": "fluid.component",
```
:horse::cow::chicken:.:cat:

colinbdclark commented on tests/framework-tests/preferences/js/AuxBuilderTests.js:
```diff
@@ -393,6 +392,70 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     });
 
     /*******************************************************************************
+<<<<<<< HEAD
```
Stray conflict marker sneakily hidden within a comment.

amb26 replied:
One feature that is completely undesirable in my 2007 JavaScript-editing Eclipse plugin is comment folding ....

colinbdclark commented on tests/framework-tests/preferences/js/PageEnhancerTests.js:
```diff
@@ -11,25 +11,24 @@ You may obtain a copy of the ECL 2.0 License and BSD License at
                 type: "fluid.pageEnhancer",
                 options: {
-                    creatOnEvent: "createPageEnhancer",
+                    creatOnEvent: "createPageEnhancer", // TODO: HORRIBLE typo here
```
Is this actually a TODO task, or is the mis-spelled createOnEvent declaration simply unnecessary?

amb26 replied:
The TODO is to determine whether it is indeed unnecessary (which one would have to imagine it is)

colinbdclark commented:
@amb26 Can you double-check all the license header changes in this pull to ensure they are correct, as we discussed in Copenhagen?

colinbdclark commented:
@michelled The functional consequences of this pull request are completely massive, so we should do some basic testing prior to merging. Can you help me coordinate it across our supported range of browser?

amb26 commented:
Thanks, @colinbdclark - all comments so far to date believed addressed

acheetham commented:
I tested the reorderer tableRow example at examples/components/reorderer/tableRow/ and got the following error:
ASSERTION FAILED: Error in assembleCreatorArguments: cannot look up component type name fluid.messageResolver to a component creator grade with an argumentMap
    while 
instantiating dependent component with name " resolver " with record Object { type="fluid.messageResolver",  options={...}}  as child of Object { typeName="fluid.reorderer.labeller",  id="4spi4rrh-52",  lifecycleStatus="constructing",  more...}
    while 
instantiating dependent components for component Object { typeName="fluid.reorderer.labeller",  id="4spi4rrh-52",  lifecycleStatus="constructing",  more...}
    while 
constructing component of type fluid.reorderer.labeller  with arguments [Object { marker={...},  mergeRecords={...},  instantiator={...},  more...}]
    while 
instantiating dependent component with name " labeller " with record Object { options={...},  type="fluid.reorderer.labeller"}  as child of Object { typeName="fluid.reorderList",  id="4spi4rrh-10",  lifecycleStatus="constructing",  more...}
    while 
instantiating dependent components for component Object { typeName="fluid.reorderList",  id="4spi4rrh-10",  lifecycleStatus="constructing",  more...}
    while 
constructing component of type fluid.reorderList  with arguments ["tbody", Object { selectors={...},  components={...}}]

acheetham commented:
Testing examples/framework/preferences/conditionalAdjusters/, I got the following error:

Tue Jul 21 2015 14:45:39 GMT-0400 (EDT):  
ASSERTION FAILED: Error in invoker record: could not resolve members func, funcName or method to a function implementation - got undefined from Object { funcName="fluid.prefs.dataSource.get"}
    while 
beginning instantiation of invoker with name get  and record fluid.prefs.dataSource.get  as child of Object { typeName="fluid.prefs.store",  id="8madegc1-77",  lifecycleStatus="constructing",  more...}
    while 
instantiating dependent components for component Object { typeName="fluid.prefs.store",  id="8madegc1-77",  lifecycleStatus="constructing",  more...}
    while 
constructing component of type fluid.prefs.store  with arguments [Object { marker={...},  mergeRecords={...},  instantiator={...},  more...}]
    while 
instantiating dependent component with name " settingsStore " with record Object { type="fluid.prefs.store",  options={...}}  as child of Object { typeName="fluid.component",  id="8madegc1-73",  lifecycleStatus="constructing",  more...}
    while 
instantiating dependent components for component Object { typeName="fluid.component",  id="8madegc1-73",  lifecycleStatus="constructing",  more...}
    while 
constructing component of type fluid.component  with arguments [Object { marker={...},  mergeRecords={...},  instantiator={...},  more...}]
    while 
instantiating dependent component with name " store " with record Object { type="fluid.component",  options={...}}  as child of Object { typeName="fluid.prefs.created_8madegc1-44.prefsEditor",  id="8madegc1-48",  lifecycleStatus="constructing",  more...}
    while 
instantiating dependent components for component Object { typeName="fluid.prefs.created_8madegc1-44.prefsEditor",  id="8madegc1-48",  lifecycleStatus="constructing",  more...}
    while 
constructing component of type fluid.prefs.created_8madegc1-44.prefsEditor  with arguments [".flc-prefsEditor-separatedPanel", undefined]

acheetham commented:
It looks like our rich text editor links are failing. Testing tests/manual-tests/components/inlineEdit/rich/, I got the following errors:

"NetworkError: 404 Not Found - http://tinymce-fluid.appspot.com/3.4.9/tiny_mce_src.js"
"NetworkError: 404 Not Found - http://ckeditor-fluid.appspot.com/ckeditor.js"

and

Tue Jul 21 2015 14:50:06 GMT-0400 (EDT):  
ASSERTION FAILED: Could not resolve reference tinyMCE to a value
    while 
firing to listener to event named onCreate  of component Object { typeName="fluid.inlineEdit.tinyMCE",  id="86iwnoui-11",  lifecycleStatus="constructed",  more...}
    while 
constructing component of type fluid.inlineEdit.tinyMCE  with arguments ["#demo-richInlineEdit-container-tinyMCE", Object { tinyMCE={...},  componentDecorators={...},  strings={...},  more...}]

acheetham commented:
Running tests/manual-tests/framework/core/multipleVersions/, I got the following errors:

Tue Jul 21 2015 14:53:29 GMT-0400 (EDT):  

ASSERTION FAILED: Cannot autoInit component fluid.component which does not have an initFunction defined

acheetham commented:
The self-voicing enactor test at tests/framework-tests/preferences/html/SelfVoicingEnactor-test.html is timing out in Safari on MacOS, with a console message of:

[Debug] Tue Jul 21 2015 15:19:48 GMT-0400 (EDT):   Test case listener has not responded after 5000ms - at sequence pos 2 of 2 sequence element 
Object
  of fixture Dom Reading

Test passes in Chrome on MacOS (and doesn't run, by design, in Firefox).

michelled commented:
The 'undo' feature of the Inline Edit component is no longer available. 
Steps to reproduce: 
1. Open demos/inlineEdit/
2. Modify the title
Note that the undo does not appear.

acheetham commented:
When loading the prefsEditor demo from http://tor1-tst-app01.inclusivedesign.ca/demos/prefsFramework/, the paths to the templates and message bundles is wrong (it's trying http://tor1-tst-app01.inclusivedesign.ca/framework/... instead of http://tor1-tst-app01.inclusivedesign.ca/src/framework/...)

michelled commented:
Data binding in the renderer example is no longer working. 

Steps to reproduce: 
1. Open demos/renderer/
2. Select the 'Render' button
3. Change the location
Note that the location does not change in the 'model' pane.

simonbates commented:
I've found an issue on IE11 on the grid reorderer demo http://tor1-tst-app01.inclusivedesign.ca/demos/reorderer/gridReorderer/

It appears to be limited to IE and I cannot reproduce on FF or Chrome.

To reproduce:

1. load the demo page
2. close the overview panel
3. tab to the letter puzzle
4. arrow key to "H"
5. press CTRL + up and then left to move "H" to the top left, release CTRL
6. press CTRL + right and right again

The bottom right square disappears and we see:

F H A
E B I
D C

It appears to be timing related. If I move slowly it works, if quickly it breaks. Also, if I have the Developer Tools open, I cannot reproduce the problem :(

michelled commented:
The layout reorderer example shows an invalid drop target warning when there is a valid drop target.

Steps to reproduce: 
   1. Open demos/reorderer/layoutReorderer/
   2. Using the mouse, pick up the 'Survey Results' panel and hover over the 'The Making of a Thneed' panel.
Note that the warning appears along with the drop target directly below the panel you are hovering over. If you release the panel, it moves to where the drop target was indicated.

michelled commented:
The table of contents does not appear.

Steps to reproduce: 
   1. Open demos/tableOfContents/

Note that there is no table of contents at the top of the page.

jobara commented:
@amb26 and @acheetham In regards to the self-voicing enactor tests timing out in Safari, I believe that is addressed by changes in the pull request for the mock text-to-speech https://github.com/fluid-project/infusion/pull/612

amb26 commented:
Hi all - I've updated this branch with fixes to all the regressions listed above which are due to the branch, Simon has JIRAed the IE11 issue at https://issues.fluidproject.org/browse/FLUID-5715 and I JIRAed the CDN issue at https://issues.fluidproject.org/browse/FLUID-5719 . Fixing the preferences demo revealed a core framework bug of https://issues.fluidproject.org/browse/FLUID-5717 which is fixed here, as well as reminding us that FLUID-5615 is still active. As for @acheetham 's issue about the paths of templates in the prefs demo, this seems to be an issue with how that build was deployed since the demo seems to work fine locally - or it may be an issue with the build process itself. For @michelled 's layoutReorderer issue I believe this is "working as designed" - the drop warning is specified to be displayed when we mouse over a locked portlet. It's trunk that has been broken in this respect, possibly for a couple of years, since this behaviour was lost at this point. Ready for another round of review and checking.

acheetham commented:
The dynamic reorderer manual test at tests/manual-tests/components/reorderer/dynamic/ is producing the following error (which is not happening in master):

Wed Jul 29 2015 10:39:01 GMT-0400 (EDT):  
ASSERTION FAILED: Error in assembleCreatorArguments: cannot look up component type name fluid.messageResolver to a component creator grade with an argumentMap
Fluid.js (line 1)
    while 
instantiating dependent component with name " resolver " with record Object { type="fluid.messageResolver",  options={...}}  as child of Object { typeName="fluid.reorderer.labeller",  id="1triwhdu-51",  lifecycleStatus="constructing",  more...}
Fluid.js (line 1)
    while 
instantiating dependent components for component Object { typeName="fluid.reorderer.labeller",  id="1triwhdu-51",  lifecycleStatus="constructing",  more...}

acheetham commented:
The error described last week for
examples/framework/preferences/conditionalAdjusters-singlePanel/
examples/framework/preferences/usingGrades/
tests/manual-tests/framework/preferences/assortedContent/
tests/manual-tests/framework/preferences/fullPage/
tests/manual-tests/framework/preferences/fullPage-schema/

is still happening (console logs pasted below).

Oddly, it is no longer happening for examples/framework/preferences/conditionalAdjusters/

Wed Jul 29 2015 10:48:44 GMT-0400 (EDT):  
ASSERTION FAILED: Error in invoker record: could not resolve members func, funcName or method to a function implementation - got undefined from Object { funcName="fluid.prefs.dataSource.get"}
Fluid.js (line 1)
    while beginning instantiation of invoker with name get  and record fluid.prefs.dataSource.get  as child of Object { typeName="fluid.prefs.store",  id="6xqe70dl-33",  lifecycleStatus="constructing",  more...}
    while instantiating dependent components for component Object { typeName="fluid.prefs.store",  id="6xqe70dl-33",  lifecycleStatus="constructing",  more...}
    while constructing component of type fluid.prefs.store  with arguments [Object { marker={...},  mergeRecords={...},  instantiator={...},  more...}]
    while instantiating dependent component with name " settingsStore " with record Object { type="fluid.prefs.store",  options={...}}  as child of Object { typeName="fluid.prefs.globalSettingsStore",  id="6xqe70dl-29",  lifecycleStatus="constructing",  more...}

jobara commented:
The [conditionalAdjusters-singlePanel](http://tor1-tst-app01.inclusivedesign.ca/examples/framework/preferences/) example is throwing an error. This prevents the example from initializing.

ASSERTION FAILED:  Error in invoker record: could not resolve members func, funcName or method to a function implementation - got undefined from  Object
funcName: "fluid.prefs.dataSource.get"
__proto__: Object

acheetham commented:
In FF, the PrefsFramework demo does not work.

The speech synthesis functionality can't be found, obviously, but the sliding panel fails to open. The only error in the console is "ReferenceError: speechSynthesis is not defined," and the prefs editor seems to get reasonably far along in its instantiation (the "show" button changes its text to the rendered "show display preferences" and when you click it, it changes to "hide") but the panel does not slide down.

jobara commented:
The UIOptions demo has a console log that says "Binding to document fluid-id-etg7fjr-96". This should probably be removed.

jobara commented:
The uploader demo throws server response errors (405, 413) when trying to perform an upload. The demo should be simulating an upload, it should not hit a real server, and should succeed at uploading any file that is allowed in the queue.

maozillah commented:
Only in IE11 on windows, tests/framework-tests/preferences/html/SelfVoicingEnactor-test.html is throwing an error for "2. the test to be spoken should have been queued correctly", the result is missing 
"},
  {
    "interrupt": false,
    "text": "Reading text from DOM"
  },
  {
    "interrupt": false,
    "text": "no image"
   }"

acheetham commented:
In Chrome on the Mac, the TextToSpeech component tests run inconsistently:

From all-tests (i.e. http://tor1-tst-app01.inclusivedesign.ca/tests/all-tests.html?testNumber=47), it fails with "Expected at least one assertion, but none were run - call expect(0) to accept zero assertions"

Directly (i.e. http://tor1-tst-app01.inclusivedesign.ca/tests/component-tests/textToSpeech/html/TextToSpeech-test.html) it passes most times, but occasionally "passes" claiming no tests were run because speech is not supported.

maozillah commented:
IE11 on windows, tests/manual-tests/framework/preferences/assortedContent
Including the "error in invoker record" error, there is also an IE11 specific error of "CSS3114: @font-face failed OpenType embedding permission check. Permission must be installable"

amb26 commented:
Thanks for the all the reports. I've addressed all of them except for the two IE11 ones from @maozillah  - do you think you could issue JIRAs for these? Thanks!
I believe that @acheetham 's TextToSpeech tests are caused by an issue I discovered after merging in @jobara 's pull request https://github.com/fluid-project/infusion/pull/612 - it looks like there is a fault in the interaction of QUnit and QUnit-composite in that their shared test state will become corrupted if the test page waits too long before queuing a test. I have refactored that test impl so that the tests start as soon as possible but the feature detect runs within the test rather than outside it - this seemed to resolve the problems at least on my machine, but it is quite worrying that the feature detect is so slow. Sorry to miss the remainder of @acheetham 's dataSource reports, I hadn't realised that there was a whole paragraph of them in the original report - these should now be all fixed.

maozillah commented:
@amb26 will do!

jobara commented:
The TextToSpeech tests will only run the "Initialization" test in Safari. The others halt. This happens both in the all-tests and when running the test individually.

maozillah commented:
for examples/framework/preferences/usingGrades/
Still getting an console message: 
"Uncaught Error: Assertion failure - check console for more details: Error in invoker record: could not resolve members func, funcName or method to a function implementation - got undefined from [object Object]"

jobara commented:
In the uploader demo, if you attempt to queue a file that exceeds the file size limit ( i believe this is over 20MB, you get a warning. However, clicking on the "Show files" link does not show a list of the files that were not queued.

![screen shot 2015-07-30 at 1 20 59 pm](https://cloud.githubusercontent.com/assets/556024/8990036/d438ad10-36bd-11e5-8845-c6994bbc8680.png)

jobara commented:
@amb26 as per our conversation in the [IRC channel](https://botbot.me/freenode/fluid-work/2015-07-30/?msg=45986338&page=2). changing the checkTTSSupport method to resolve after onstop instead of onstart seems to address the test issue in Safari.

amb26 commented:
Thanks for the further testing - all issues so far reported should be fixed.

cindyli commented:
In Chrome 44 on Mac 10.10.4, loading tests/manual-tests/components/inlineEdit/rich/ throws error:

ASSERTION FAILED:  Could not resolve reference tinyMCE to a value

while  firing to listener to event named  onCreate  of component  Object {typeName: "fluid.inlineEdit.tinyMCE", id: "5pz8yf-11", lifecycleStatus: "constructed", options: Object, events: Object…}applier: fluid.ChangeAppliercancel: ()container: jQuery.fn.init[1]decorators: Array[1]destroy: ()displayModeRenderer: jQuery.fn.init[1]displayView: Objectdom: Objectedit: ()editContainer: jQuery.fn.init[1]editField: jQuery.fn.init[1]events: ObjectexistingPadding: 0finish: ()id: "5pz8yf-11"isEditing: ()isEditingState: falselifecycleStatus: "constructed"locate: (name,localContainer)model: ObjectmodelRelay: nulloptions: ObjectrefreshView: ()textEditButton: jQuery.fn.init[1]tooltipEnabled: ()typeName: "fluid.inlineEdit.tinyMCE"undo: ObjectupdateModel: ()updateModelValue: ()viewEl: jQuery.fn.init[1]__proto__: fluid.componentConstructor

while  constructing component of type  fluid.inlineEdit.tinyMCE  with arguments  ["#demo-richInlineEdit-container-tinyMCE", Object]

amb26 commented:
Thanks @cindyli - this issue is already reported at https://issues.fluidproject.org/browse/FLUID-5719
